src/lib/Lib/Filters.hpp

```diff
@@ -0,0 +1,98 @@
+//
+// This is a derivative work. originally part of the LLVM Project.
+// Licensed under the Apache License v2.0 with LLVM Exceptions.
+// See https://llvm.org/LICENSE.txt for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+//
+// Copyright (c) 2023 Krystian Stasiowski (sdkrystian@gmail.com)
+//
+// Official repository: https://github.com/cppalliance/mrdox
+//
+
+#ifndef MRDOX_LIB_FILTERS_HPP
+#define MRDOX_LIB_FILTERS_HPP
+
+#include <mrdox/Platform.hpp>
+#include <span>
+#include <string>
+#include <string_view>
+#include <vector>
+
+namespace clang {
+namespace mrdox {
+
+class FilterPattern
+{
+    // pattern without any wildcards
+    std::string raw_;
+    // pattern part lengths, where zero represents a wildcard
+    std::vector<std::size_t> parts_;
+
+#ifndef NDEBUG
+    std::string pattern_ = "*";
+#endif
+
+    bool
+    matchesSlow(
```

Comment:
Could you please add some comments to the `matchesSlow` function? It'd help clarify its logic and purpose for anyone reading through the code in the future.